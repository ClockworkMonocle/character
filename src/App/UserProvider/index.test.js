import React from 'react';
import { shallow } from 'enzyme';
import { UserProvider } from './index';

test('<UserProvider /> renders loading when loading user', () => {
  expect(
    shallow(
      <UserProvider
        user={{ isAuthenticating: true }}
        authenticateUser={() => {}}
      >
        <span>hey</span>
      </UserProvider>
    )
  ).toMatchSnapshot();
});

test('<UserProvider /> renders children when user is loaded', () => {
  expect(
    shallow(
      <UserProvider
        user={{ isAuthenticating: false }}
        authenticateUser={() => {}}
      >
        <span>hey</span>
      </UserProvider>
    )
  ).toMatchSnapshot();
});

test('<UserProvider /> renders error if problem loading user', () => {
  expect(
    shallow(
      <UserProvider
        user={{ error: 'Something went wrong...' }}
        authenticateUser={() => {}}
      >
        <span>hey</span>
      </UserProvider>
    )
  ).toMatchSnapshot();
});
