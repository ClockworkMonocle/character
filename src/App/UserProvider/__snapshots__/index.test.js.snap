// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProvider /> renders children when user is loaded 1`] = `
<span>
  hey
</span>
`;

exports[`<UserProvider /> renders error if problem loading user 1`] = `
<section
  className="user_provider__container"
>
  <Type
    color="black"
    element="h1"
    variant="display-1"
  >
    Some error occurred
  </Type>
  <Type
    color="black"
    element="p"
    variant="body"
  >
    <Link
      replace={false}
      to="/login"
    >
      Try logging in again
    </Link>
  </Type>
</section>
`;

exports[`<UserProvider /> renders loading when loading user 1`] = `
<section
  className="user_provider__container"
>
  <LoadingIndicator
    color="attack-color"
  />
</section>
`;
